framework/src/modules/nft/cc_commands/cc_transfer.ts

```diff
@@ -105,24 +105,24 @@ export class CrossChainTransferCommand extends BaseCCCommand {
 
 		if (nftChainID.equals(ownChainID)) {
 			const storeData = await nftStore.get(getMethodContext(), nftID);
+
 			if (status === CCM_STATUS_CODE_OK) {
-				storeData.owner = recipientAddress;
 				const storedAttributes = storeData.attributesArray;
 				storeData.attributesArray = this._internalMethod.getNewAttributes(
 					nftID,
 					storedAttributes,
 					receivedAttributes,
 				);
-				await nftStore.save(getMethodContext(), nftID, storeData);
-				await this._internalMethod.createUserEntry(getMethodContext(), recipientAddress, nftID);
-				await escrowStore.del(getMethodContext(), escrowStore.getKey(sendingChainID, nftID));
-			} else {
+			}
+
+			if (status !== CCM_STATUS_CODE_OK) {
 				recipientAddress = senderAddress;
-				storeData.owner = recipientAddress;
-				await nftStore.save(getMethodContext(), nftID, storeData);
-				await this._internalMethod.createUserEntry(getMethodContext(), recipientAddress, nftID);
-				await escrowStore.del(getMethodContext(), escrowStore.getKey(sendingChainID, nftID));
 			}
+
+			storeData.owner = recipientAddress;
+			await nftStore.save(getMethodContext(), nftID, storeData);
```

Comment:
> In particular, two places may result in the NFT having duplicate module attributes
When a foreign NFT is received (lisk-sdk/framework/src/modules/nft/cc_commands/cc_transfer.ts#L142-L145)
When a foreign NFT is bounced (lisk-sdk/framework/src/modules/nft/cc_commands/cc_transfer.ts#L149-L152)

As per this point in the issue; to avoid duplicate attributes, perhaps createNFTEntry is to be called here as well?! 🤔  [LIP's definition of the method](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md#execution-2)